### PR TITLE
Prefs [Task][27979]: Manage fragment

### PR DESCRIPTION
**Pull request by NguyenTaTrung**
## Related Tickets
- [#Task_27979: Manage fragment](https://edu-redmine.sun-asterisk.vn/issues/27979)

## WHAT
- Control fragment
- Add animaton when create or remove fragment

## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Trainer reivew (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the source code have end-line errors? | | NO |<li>- [x] no</li>|<li>- [ ] no</li>  
Conventions | No hard coding, use constants/configuration values | | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>

### CODE REVIEW

File: app/src/main/res/anim/fragment_exit_to_right.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<set xmlns:android="http://schemas.android.com/apk/res/android"
+    android:shareInterpolator="false">
+    <translate
+        android:fromXDelta="0%" android:toXDelta="100%p"
```
Comment (huytq-0932) [inline app/src/main/res/anim/fragment_exit_to_right.xml:5]: %p là sao nhỉ

File: app/src/main/java/com/sunasterisk/anticovid_19/utils/FragmentUtil.kt [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+    fun addAdditionalTabFragment(
+        fragmentManager: FragmentManager,
+        tagStacks: Map<String, Stack<String>>,
+        tag: String,
+        show: Fragment,
+        hide: Fragment,
+        layoutId: Int,
+        shouldAddToStack: Boolean
+    ) {
```
Comment (thonc-2135) [inline app/src/main/java/com/sunasterisk/anticovid_19/utils/FragmentUtil.kt:36]: tận 7 tham số, có nhiều quá k, thg chỉ đến 3 parametter là cùng thôi

File: app/src/main/java/com/sunasterisk/anticovid_19/utils/FragmentUtil.kt [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        action: String,
+        tab: String?,
+        shouldAdd: Boolean,
+        fragmentInteractionCallback: BaseFragment.FragmentInteractionCallback?
+    ) {
+        val bundle = Bundle()
+        bundle.putString(BUNDLE_ACTION, action)
+        bundle.putString(BUNDLE_NAME_TAB, tab)
+        bundle.putBoolean(BUNDLE_SHOULD_ADD, shouldAdd)
+        fragmentInteractionCallback?.onFragmentInteractionCallBack(bundle)
```
Comment (thonc-2135) [inline app/src/main/java/com/sunasterisk/anticovid_19/utils/FragmentUtil.kt]: dùng scope function cho gọn nhé

File: app/src/main/java/com/sunasterisk/anticovid_19/utils/Constants.kt [hunk 4]
```diff
@@ -43,3 +43,9 @@ object PreferencesConst {
     const val PREFS_IS_ALLOW_NOTIFICATION = "PREFS_IS_ALLOW_NOTIFICATION"
```
Comment (thonc-2135) [inline app/src/main/java/com/sunasterisk/anticovid_19/utils/Constants.kt:43]: tưởng "PREFS_............." là viết thường nhỉ ?

File: app/src/main/java/com/sunasterisk/anticovid_19/utils/FragmentUtil.kt (hunk 2 above)
Comment (NguyenTaTrung) [reply app/src/main/java/com/sunasterisk/anticovid_19/utils/FragmentUtil.kt:36]: Cần ông để xử backstack ý mà

File: app/src/main/java/com/sunasterisk/anticovid_19/utils/FragmentUtil.kt (hunk 3 above)
Comment (NguyenTaTrung) [reply app/src/main/java/com/sunasterisk/anticovid_19/utils/FragmentUtil.kt]: oki

File: app/src/main/java/com/sunasterisk/anticovid_19/utils/Constants.kt (hunk 4 above)
Comment (NguyenTaTrung) [reply app/src/main/java/com/sunasterisk/anticovid_19/utils/Constants.kt:43]: t flow theo convention của công ty mà mục 2.10
https://github.com/framgia/coding-standards/blob/master/eng/android/codingstyleguide.md

**Review by tuanpa-0908 — APPROVED** [review]

**Pull request merged**